python/tank/authentication/user_impl.py

```diff
@@ -53,11 +53,11 @@ def __init__(self, host, http_proxy):
         # we don't end up infecting API instances with unicode strings
         # that would then cause some string data to be unicoded during
         # concatenation operations.
-        if isinstance(http_proxy, unicode):
-            http_proxy = http_proxy.encode("utf-8")
+        if http_proxy is not None:
```

Comment:
Shouldn't this be?
```suggestion
        if isinstance(http_proxy, six.text_type):
```

Reply:
So previously this was encoding unicode strings, and leaving unicode as is.  We need to treat this differently on python 2/3, and six has a helper function for this exact use (`ensure_str`) -- the only reason we keep the `is not None` test at all is that function will only accept `str`, `unicode`, `bytes`, but http_proxy can be `None` which will raise an exception if passed to `ensure_str`.

Reply:
Ok, thanks for the clarification.